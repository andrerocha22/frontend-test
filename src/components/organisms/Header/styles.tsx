import styled from 'styled-components';

export const HeaderContainer = styled.header`
  position: absolute;
  top: 0;
  width: 100%;
  background-color: #ffffff;
  padding: 2.8rem 3.7rem 1.9rem;
  z-index: 10001;

  @media all and (max-width: 768px) {
    padding: 2.1rem 1.6rem 1.4rem;
  }
`;

export const Logo = styled.img`
  @media all and (max-width: 768px) {
    width: 6.5rem;
  }
`;
