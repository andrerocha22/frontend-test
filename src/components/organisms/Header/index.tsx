import React from 'react';
import logo from '../../../assets/images/logoPrimary.png';
import { HeaderContainer, Logo } from './styles';

const Header: React.FC = () => {
  return (
    <HeaderContainer data-test="component-header">
      <a href="/">
        <Logo className="logoIMG" src={logo} alt="Logo-Origin" />
      </a>
    </HeaderContainer>
  );
};

export default Header;
