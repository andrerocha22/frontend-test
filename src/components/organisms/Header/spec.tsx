import React from 'react';
import { shallow } from 'enzyme';
import { findByTestAttr } from '../../../test/testUtils';
import Header from './index';

const setup = (props = {}) => {
  const setupProps = { ...props };
  return shallow(<Header {...setupProps} />).dive();
};

describe('Header component', () => {
  test('renders without error', () => {
    const wrapper = setup();
    const component = findByTestAttr(wrapper, 'component-header');
    expect(component.length).toBe(1);
  });

  test('goal hero snapshot test ', () => {
    const wrapper = setup();
    const component = findByTestAttr(wrapper, 'component-header');
    expect(component).toMatchSnapshot();
  });
});
