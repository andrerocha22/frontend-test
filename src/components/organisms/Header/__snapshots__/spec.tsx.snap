// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component goal hero snapshot test  1`] = `
<header
  className="sc-bdnylx jiINeO"
  data-test="component-header"
>
  <a
    href="/"
  >
    <styled.img
      alt="Logo-Origin"
      className="logoIMG"
      src="test-file-stub"
    />
  </a>
</header>
`;
